### ISSUE TITLE
Error raised when launching the app: Specify either app or appTopLevelWindow to create a session

### ISSUE THREAD

**Issue by bwomsm1** [microsoft/WinAppDriver#1774]
I am trying to launch my test with the following capabilities: 
```
{
  "capabilities": {
    "firstMatch": [
      {
        
      }
    ],
    "alwaysMatch": {
      "appium:app": "C:\\AeroScoutEngineManager\\bin\\Rte\\EngineManager.exe",
      "appium:winAppDriverHost": "127.0.0.1:4723",
      "platformName": "Windows",
      "appium:platformVersion": "10",
      "appium:automationName": "Windows",
      "appium:appArguments": "ShellLayout.xml ShellCnf.xml",
      "appium:appWorkingDir": "c:\\AeroScoutEngineManager\\bin\\Rte"
    }
  }
}
```

My application must be launched with the arguments above. from that specific path.

The error I am getting is:
```
Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.


==========================================
POST /session HTTP/1.1

Accept: application/json

Accept-Encoding: identity

Connection: keep-alive

Content-Length: 387

Content-Type: application/json;charset=UTF-8

Host: 127.0.0.1:4723

User-Agent: appium/python 2.6.0 (selenium/4.3.0 (python windows))

X-Idempotency-Key: 4479771e-d340-492a-9486-c96fe20d15c2



{"capabilities": {"firstMatch": [{}], "alwaysMatch": {"appium:app": "C:\\AeroScoutEngineManager\\bin\\Rte\\EngineManager.exe", "appium:winAppDriverHost": "127.0.0.1:4723", "platformName": "Windows", "appium:platformVersion": "10", "appium:automationName": "Windows", "appium:appArguments": "ShellLayout.xml ShellCnf.xml", "appium:appWorkingDir": "c:\\AeroScoutEngineManager\\bin\\Rte"}}}
HTTP/1.1 400 Bad Request

Content-Length: 141

Content-Type: application/json



{"status":100,"value":{"error":"invalid argument","message":"Bad capabilities. Specify either app or appTopLevelWindow to create a session"}}


==========================================
POST /session HTTP/1.1

Accept: application/json

Accept-Encoding: identity

Connection: keep-alive

Content-Length: 387

Content-Type: application/json;charset=UTF-8

Host: 127.0.0.1:4723

User-Agent: appium/python 2.6.0 (selenium/4.3.0 (python windows))

X-Idempotency-Key: e71805eb-5c48-40b8-8a38-c45b801d244d



{"capabilities": {"firstMatch": [{}], "alwaysMatch": {"appium:app": "C:\\AeroScoutEngineManager\\bin\\Rte\\EngineManager.exe", "appium:winAppDriverHost": "127.0.0.1:4723", "platformName": "Windows", "appium:platformVersion": "10", "appium:automationName": "Windows", "appium:appArguments": "ShellLayout.xml ShellCnf.xml", "appium:appWorkingDir": "c:\\AeroScoutEngineManager\\bin\\Rte"}}}
HTTP/1.1 400 Bad Request

Content-Length: 141

Content-Type: application/json



{"status":100,"value":{"error":"invalid argument","message":"Bad capabilities. Specify either app or appTopLevelWindow to create a session"}}
==========================================
```

What is the problem?

Thanks!

**Comment by bwomsm1** [comment]
[](url)[](url
[win_app_driver.log](https://github.com/microsoft/WinAppDriver/files/9307527/win_app_driver.log)
)

**Comment by anunay1** [comment]
Are you using the latest release of Appium? if yes then you need to downgrade as WAD is not w3c compliant yet.

**Comment by bideee** [comment]
> Are you using the latest release of Appium? if yes then you need to downgrade as WAD is not w3c compliant yet.

Can it work with Selenium v4? I've upgraded to latest version of Appium in a way for it to work with Selenium v4.

**Comment by anunay1** [comment]
no it will not  work with selenium4.

**Comment by bwomsm1** [comment]
@anunay1 I am bit confused, I am running with appium version `1.22.3` and these are my python libs:

```
setuptools==63.2.0
Appium-Python-Client==2.6.0
configparser==5.2.0
requests==2.28.1
pyserial==3.5
robotframework==5.0.1
selenium==4.3.0
robotbackgroundlogger==1.2
robotframework-metrics==3.3.3
webdriver-manager==3.8.1
```

What lib version is wrong?

**Comment by anunay1** [comment]
The selenium version should be 3.141.1

**Comment by anunay1** [comment]
Also please check https://github.com/anunay1/SampleWADPython

**Comment by bwomsm1** [comment]
@anunay1 I downgraded to selenium 3.141.11 but I am still facing the same error

**Comment by bwomsm1** [comment]
I am using Appium `1.22.3`

**Comment by bwomsm1** [comment]
@anunay1 I really need your advice, I downgraded to selenium 3.141.1 but I am still facing the same error.

**Comment by anunay1** [comment]
Also does your application have splash screen

**Comment by bwomsm1** [comment]
@anunay1 No, I don't have a splash screen, I must say that I am getting the same error when trying to launch notepad.

**Comment by anunay1** [comment]
Can we have a teams meeting?

**Comment by bwomsm1** [comment]
@anunay1 sure, what about zoom: https://stanleyblackanddecker.zoom.us/j/97591280882?

I am there.

**Comment by Shilpiiiii** [comment]
I am also facing the same issue, but as mentioned above, I have downgraded the Appium-python-client lib version to 2.2.0 but for the selenium, I couldn't find the 3.141.1 version.

**Comment by anunay1** [comment]
How are you trying to downgrade?

**Comment by Shilpiiiii** [comment]
uninstalled the previously installed selenium version by pip uninstall selenium and then I am trying to install the selenium 3.141.1 version using pip install selenium==3.141.1 but getting this error - ERROR: Could not find a version that satisfies the requirement selenium==3.141.1

**Comment by anunay1** [comment]
the version is 3.141.0 , use this.  
<img width="711" alt="image" src="https://user-images.githubusercontent.com/7989610/185769838-f6e62b78-bf7d-4246-890e-9104a69d0651.png">

**Comment by Shilpiiiii** [comment]
Hey, I am trying to launch outlook using winappdriver but i am not able to do that. I am new to python and winappdriver both. not sure what's wrong. i ran the calculator program and its working fine but not any Office app. @anunay1 can you please help me

**Comment by anunay1** [comment]
> Hey, I am trying to launch outlook using winappdriver but i am not able to do that. I am new to python and winappdriver both. not sure what's wrong. i ran the calculator program and its working fine but not any Office app. @anunay1 can you please help me

What's the error?

**Comment by Shilpiiiii** [comment]
selenium.common.exceptions.WebDriverException: Message: Bad capabilities. Specify either app or appTopLevelWindow to create a session

**Comment by bwomsm1** [comment]
@Shilpiiiii what are your python versions of `Appium-Python-Client` and `selenium`? from my experience with that error you need to downgrade these libs versions

**Comment by samuelfreiberg** [comment]
@anunay1 After reading this thread, I want to make sure I understand correctly...

WinAppDriver is not W3C compliant yet? And due to this, there might be gaps in API with new versions of Appium? Our team is currently on Appium `"appium": "2.0.0-beta.41"` due to some compliance issues with packages that Appium v1 brought in.

**Comment by bjanzen** [comment]
I got - "appium-python-client 2.2.0 requires selenium~=4.1, but you have selenium 3.14.1 which is incompatible." ??

> Problem was related to Appium-Python-Client lib version, I had to downgrade from `2.6.0` to `2.2.0` and the issue solved. In addition to downgrade Selenium version `3.141.1`
> 
> (Thanks @anunay1)

**Comment by bjanzen** [comment]
Answering my own question:  I'm on Win11 so this only works on Win10 as well.

**Comment by SanthoshKumar3624** [comment]
ValueError: Timeout value connect was <object object at 0x000002957C3E6160>, but it must be an int, float or None.


For the below versions


Package              Version
-------------------- -----------
Appium-Python-Client 2.2.0
attrs                23.2.0
certifi              2024.2.2
cffi                 1.16.0
colorama             0.4.6
exceptiongroup       1.2.0
h11                  0.14.0
idna                 3.7
iniconfig            2.0.0
outcome              1.3.0.post0
packaging            24.0
pip                  21.3.1
pluggy               1.4.0
pycparser            2.22
PySocks              1.7.1
pytest               8.1.1
selenium             3.141.0
setuptools           60.2.0
sniffio              1.3.1
sortedcontainers     2.4.0
tomli                2.0.1
trio                 0.25.0
trio-websocket       0.11.1
typing_extensions    4.11.0
urllib3              2.2.1
wheel                0.37.1
wsproto              1.2.0

**Comment by SanthoshKumar3624** [comment]
wirting back to the same, downgrade urllib3  to 1.26.16  along with the changes mentioned earlier

**Comment by anunay1** [comment]
Share the code

**Comment by amitbhu61** [comment]
I'm getting the error >  Failed to initialize the driver: Message: Bad capabilities. Specify either app or appTopLevelWindow to create a session. 
 I'm trying to launched notepad. 

Name: Appium-Python-Client
Version: 4.4.0
Summary: Python client for Appium
Home-page: http://appium.io/
Author: Isaac Murchie
Author-email: isaac@saucelabs.com
License: Apache 2.0
Location: C:\Users\Dell\AppData\Local\Programs\Python\Python313\Lib\site-packages
Requires: selenium
Required-by:
---
Name: selenium
Version: 4.27.1
Summary: Official Python bindings for Selenium WebDriver
Home-page: https://www.selenium.dev
Author:
Author-email:
License: Apache 2.0
Location: C:\Users\Dell\AppData\Local\Programs\Python\Python313\Lib\site-packages
Requires: certifi, trio, trio-websocket, typing_extensions, urllib3, websocket-client
Required-by: Appium-Python-Client @anunay1 @bwomsm1  @samuelfreiberg  Can anyone help me

**Comment by anunay1** [comment]
Selenium 4 is not supported move to selenium 3.

**Comment by amitbhu61** [comment]
What about Appium-Python-Client version? which version and setuptools version=? 
@anunay1

**Comment by anunay1** [comment]
2.2.0

**Comment by Anu88680** [comment]
@anunay1 @bwomsm1 I have dowgraded both Appium-Python-client 2.2.0 and Selenium to 3.141.0. Still, i am getting the following error.

Traceback (most recent call last):
  File "C:\Users\HXXXXX\PycharmProjects\pythonProject1\opendesktopapp.py", line 12, in <module>
    driver = webdriver.Remote(
             ^^^^^^^^^^^^^^^^^
TypeError: WebDriver.__init__() got an unexpected keyword argument 'desired_capabilities'

**Comment by anunay1** [comment]
Code ????

**Comment by Anu88680** [comment]
from appium import webdriver
from time import sleep

# Define capabilities
desired_caps = {
    "platformName": "Windows",
    "deviceName": "WindowsPC",
    "app": "notepad.exe"
}

# Connect to WinAppDriver
driver = webdriver.Remote(
    command_executor="http://127.0.0.1:4723",
    desired_capabilities=desired_caps
)

# Wait for Notepad to open
sleep(2)

# Find the text editor and type text
editor = driver.find_element("class name", "Edit")
editor.send_keys("Hello from Python + WinAppDriver!")

# Wait to see the result
sleep(3)

# Close Notepad
driver.quit()

**Comment by anunay1** [comment]
Code is okay. Issue is with the selenium library it has not been downgraded successfully

**Comment by anunay1** [comment]
Are you new to contribution in open source???

**Comment by Anu88680** [comment]
@anunay1 Name: selenium
Version: 3.141.0
Summary: Python bindings for Selenium
Home-page: https://github.com/SeleniumHQ/selenium/
Author: UNKNOWN
Author-email: UNKNOWN
License: Apache 2.0
Location: C:\Users\HXXXX\AppData\Local\Programs\Python\Python312\Lib\site-packages
Requires: urllib3
Required-by: Appium-Python-Client

This is current version of selenium. I also got the following conflict message when I downgraded the selenium version.

C:\Users\HXXXXX>pip install selenium==3.141.0
Collecting selenium==3.141.0
  Using cached selenium-3.141.0-py2.py3-none-any.whl.metadata (6.6 kB)
Requirement already satisfied: urllib3 in c:\users\h304110\appdata\local\programs\python\python312\lib\site-packages (from selenium==3.141.0) (2.4.0)
Using cached selenium-3.141.0-py2.py3-none-any.whl (904 kB)
Installing collected packages: selenium
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
appium-python-client 2.2.0 requires selenium~=4.1, but you have selenium 3.141.0 which is incompatible.
Successfully installed selenium-3.141.0
 
Sorry I had to delete my previous comments that have my phone number.
